code/game/machinery/autolathe/matterforge.dm

```diff
@@ -0,0 +1,606 @@
+		eating = user.get_active_hand()
+	if(!istype(eating))
+		return FALSE
+	if(stat)
+		return FALSE
+	if(stored_material[MATERIAL_COMPRESSED_MATTER] == storage_capacity)
+		return FALSE
+	if(!Adjacent(user) && !Adjacent(eating))
+		return FALSE
+	if(!length(eating.get_matter()))
+		to_chat(user, SPAN_WARNING("\The [eating] does not contain significant amounts of useful materials and cannot be accepted."))
+		return FALSE
+	var/total_used = 0     // Amount of material used.
+	var/mass_per_sheet = 0 // Amount of material constituting one sheet.
+	var/list/total_material_gained = list()
+	for(var/obj/O in eating.GetAllContents(includeSelf = TRUE))
+		var/list/_matter = O.get_matter()
+		if(_matter)
+			for(var/material in _matter)
+				if(material in unsuitable_materials)
+					continue
+
+				if(!(material in total_material_gained))
+					total_material_gained[material] = 0
+
+				var/total_material = _matter[material]
+				if(istype(O, /obj/item/stack))
+					var/obj/item/stack/material/stack = O
+					total_material *= stack.get_amount()
+
+				if(istype(O, /obj/item/stack/material/compressed_matter))
+					to_chat(user, SPAN_NOTICE("You deposit [total_material] compressed matter into \the [src]."))
+					stored_material[MATERIAL_COMPRESSED_MATTER] += total_material
+					update_desc(stored_material[MATERIAL_COMPRESSED_MATTER])
+					qdel(eating)
+					return
+				total_material_gained[material] += total_material
+				total_used += total_material
+				mass_per_sheet += O.matter[material]
+	var/datum/component/inspiration/artifact = power_source.GetComponent(/datum/component/inspiration)
+	var/gained_mats = 0
+	for(var/mat in total_material_gained)
+		var/added_mats = artifact.power * total_material_gained[mat] * lst[mat]
+		if(added_mats + stored_material[MATERIAL_COMPRESSED_MATTER] > storage_capacity)
+			added_mats = storage_capacity - stored_material[MATERIAL_COMPRESSED_MATTER]
+		var/leftover_mats = (added_mats + stored_material[MATERIAL_COMPRESSED_MATTER]) - storage_capacity
+		stored_material[MATERIAL_COMPRESSED_MATTER] += added_mats
+		update_desc(stored_material[MATERIAL_COMPRESSED_MATTER])
+		gained_mats += added_mats
+		if(leftover_mats == 0)
+			used_sheets = (added_mats / artifact.power) / lst[mat]
+		else
+			used_sheets = total_material_gained[mat]
+	if(istype(eating, /obj/item/stack))
+		var/obj/item/stack/stack = eating
+		to_chat(user, SPAN_NOTICE("You create [gained_mats] Compressed Matter from [stack.singular_name]\s in the [src]."))
+
+		if(!stack.use(used_sheets))
+			qdel(stack)	// Protects against weirdness
+	else
+		to_chat(user, SPAN_NOTICE("You recycle \the [eating] in \the [src]."))
+		qdel(eating)
+
+/obj/machinery/matter_nanoforge/proc/queue_design(datum/design/picked_design, amount=1)
+	if(!picked_design || !amount)
+		return
+
+	while(amount && queue.len < queue_max)
+		queue.Add(picked_design)
+		amount--
+
+	if(!current_design)
+		next_file()
+
+/obj/machinery/matter_nanoforge/proc/clear_queue()
```

Comment:
what is this proc for?

Reply:
this is necessary to clear the queue of building stuff

Reply:
ok, but you don't use it at any time.